### PR TITLE
Fixing core code 

### PR DESCRIPTION
**Pull request by hoangphanusc**
- Deleted primary keys insert into's
- Added SQL parameterized queries
- Added config struct

### CODE REVIEW

File: config.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+	"PostgresAdr": "postgres://postgres:postgres5263@127.0.0.1:5432/gameTrackerDb?sslmode=disable"
```
Comment (trietphm) [inline config.json]: This format is easier to modify

```
user:...
password:...
database_name:...
...
```

File: interfaces/repositories.go [hunk 2]
```diff
@@ -73,6 +77,38 @@ func (repo *DbUserRepo) FindById(id int) (usecases.User, error) {
...
+	row, _ := repo.dbHandler.Query("SELECT user_name FROM users")
+	count := 0
+	for row.Next() {
+		count++
+	}
+	return count
+}
+
+func (repo *DbUserRepo) NameExisted(userName string) bool {
+	row, _ := repo.dbHandler.Query(`SELECT user_name FROM users
```
Comment (trietphm) [inline interfaces/repositories.go]: Ignore error?

File: interfaces/repositories.go [hunk 3]
```diff
@@ -73,6 +77,38 @@ func (repo *DbUserRepo) FindById(id int) (usecases.User, error) {
 	return user, nil
 }
 
+func (repo *DbUserRepo) Count() int {
+	row, _ := repo.dbHandler.Query("SELECT user_name FROM users")
```
Comment (trietphm) [inline interfaces/repositories.go]: Ignore error?

File: interfaces/repositories.go [hunk 4]
```diff
@@ -73,6 +77,38 @@ func (repo *DbUserRepo) FindById(id int) (usecases.User, error) {
...
+func (repo *DbUserRepo) StoreInfo(user usecases.User, info string) error {
+	_, err := repo.dbHandler.Execute(`UPDATE users SET personal_info=$1
+		WHERE id=$2`, info, user.Id)
+	return err
+}
+
+func (repo *DbUserRepo) LoadInfo(user usecases.User) (string, error) {
+	row, err := repo.dbHandler.Query(`SELECT personal_info FROM users WHERE id=$1`, user.Id)
+	if err != nil {
+		return "", nil
```
Comment (trietphm) [inline interfaces/repositories.go]: Should return `error`, should not put business logic in `repository`

File: interfaces/repositories.go [hunk 5]
```diff
@@ -112,14 +146,14 @@ func NewDbLibraryRepo(dbHandlers map[string]DbHandler) *DbLibraryRepo {
 }
 
 func (repo *DbLibraryRepo) Store(library usecases.Library) error {
-	err := repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO libraries (id, player_id) VALUES ('%d', '%v')`,
-		library.Id, library.Player.Id))
+	_, err := repo.dbHandler.Execute(`INSERT INTO libraries (player_id) VALUES ($1)`,
+		library.Player.Id)
 	if err == nil {
```
Comment (trietphm) [inline interfaces/repositories.go]: `if err == nil` or `if err != nil`? You want to return here?

File: interfaces/repositories.go [hunk 6]
```diff
@@ -112,14 +146,14 @@ func NewDbLibraryRepo(dbHandlers map[string]DbHandler) *DbLibraryRepo {
...
 		return err
 	}
-	for _, item := range library.Games {
-		err = repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO gamesInLib (game_id, library_id)
-			VALUES ('%d', '%d')`, item.Id, library.Id))
+	for _, game := range library.Games {
+		_, err = repo.dbHandler.Execute(`INSERT INTO gamesInLib (game_id, library_id)
+			VALUES ($1, $2)`, game.Id, library.Id)
 		if err == nil {
```
Comment (trietphm) [inline interfaces/repositories.go]: `if err == nil` or `if err != nil`? You want to return here?

File: interfaces/repositories.go [hunk 7]
```diff
@@ -148,8 +182,8 @@ func (repo *DbLibraryRepo) FindById(id int) (usecases.Library, error) {
 
 	var gameId int
 	gameRepo := NewDbGameRepo(repo.dbHandlers)
-	row, err = repo.dbHandler.Query(fmt.Sprintf(`SELECT game_id FROM gamesInLib
-		WHERE library_id = '%d'`, library.Id))
+	row, err = repo.dbHandler.Query(`SELECT game_id FROM gamesInLib
+		WHERE library_id = $1`, library.Id)
 	if err == nil {
```
Comment (trietphm) [inline interfaces/repositories.go]: `if err == nil` or `if err != nil`? You want to return here?

File: interfaces/repositories.go (hunk 4 above)
Comment (hoangphanusc) [inline interfaces/repositories.go]:
```
func (repo *DbUserRepo) NameExisted(userName string) bool {
    row, _ := repo.dbHandler.Query(`SELECT user_name FROM users
        WHERE user_name=$1 LIMIT 1`, userName)
    return row.Next()
}
```

Are you referring to this part?

File: interfaces/repositories.go [hunk 8]
```diff
@@ -175,17 +209,17 @@ func NewDbGameRepo(dbHandlers map[string]DbHandler) *DbGameRepo {
 }
 
 func (repo *DbGameRepo) Store(game domain.Game) error {
-	err := repo.dbHandler.Execute(fmt.Sprintf(`INSERT INTO games (id, name, producer value)
-    	VALUES ('%d', '%s', '%s', '%f')`, game.Id, game.Name, game.Producer, game.Value))
+	_, err := repo.dbHandler.Execute(`INSERT INTO games (game_name, producer, value)
+    	VALUES ($1, $2, $3)`, game.Name, game.Producer, game.Value)
 	if err == nil {
```
Comment (trietphm) [inline interfaces/repositories.go]: `if err == nil` or `if err != nil`?

File: interfaces/repositories.go [hunk 9]
```diff
@@ -175,17 +209,17 @@ func NewDbGameRepo(dbHandlers map[string]DbHandler) *DbGameRepo {
...
 	}
 	return nil
 }
 
 func (repo *DbGameRepo) FindById(id int) (domain.Game, error) {
-	row, err := repo.dbHandler.Query(fmt.Sprintf(`SELECT name, producer, value
-    	FROM items WHERE id = '%d' LIMIT 1`, id))
+	row, err := repo.dbHandler.Query(`SELECT game_name, producer, value FROM items
+    	WHERE id = $1 LIMIT 1`, id)
 	if err == nil {
```
Comment (trietphm) [inline interfaces/repositories.go]: `if err == nil` or `if err != nil`?

File: interfaces/repositories.go (hunk 4 above)
Comment (trietphm) [inline interfaces/repositories.go]:
```
if err != nil {
        return "", nil
    }
```

It should be 

```
if err != nil {
        return "", err
    }
```

Comment (hoangphanusc) [inline interfaces/repositories.go]: What about the "business code" part?

Comment (trietphm) [inline interfaces/repositories.go]: Well, I mean logic.

For example:

```
row, err := repo.dbHandler.Query(`SELECT personal_info FROM users WHERE id=$1`, user.Id)
    if err != nil {
        return "", nil
}
```

If you don't care about `error`, your logic is: if error => user not found => Do some stuff

But that logic `if error => user not found => Do some stuff` should do in other layer not repository, may be `use case` or somewhere. It should return everything, especially `error`.

Comment (hoangphanusc) [inline interfaces/repositories.go]: Oh okay I see, that was just a mistake. Changed to `return "", err`

File: interfaces/repositories.go [hunk 10]
```diff
@@ -73,6 +77,41 @@ func (repo *DbUserRepo) FindById(id int) (usecases.User, error) {
 	return user, nil
 }
 
+func (repo *DbUserRepo) Count() (int, error) {
+	row, err := repo.dbHandler.Query("SELECT user_name FROM users")
```
Comment (trietphm) [inline interfaces/repositories.go]: Use this instead 
http://www.w3schools.com/sql/sql_func_count.asp

**Pull request merged**